OpenRA.Mods.Common/UpdateRules/Rules/DefineSoundDefaults.cs

```diff
@@ -0,0 +1,114 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2018 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System.Collections.Generic;
+using System.Linq;
+
+namespace OpenRA.Mods.Common.UpdateRules.Rules
+{
+	public class DefineSoundDefaults : UpdateRule
+	{
+		public override string Name { get { return "Move mod-specific sound defaults to yaml"; } }
+		public override string Description
+		{
+			get
+			{
+				return "Mod-specific default sound values have been removed from several traits.\n" +
+					"The original values are added back via yaml.";
+			}
+		}
+
+		List<MiniYamlNode> paradropNodes = new List<MiniYamlNode>();
+		List<MiniYamlNode> ejectNodes = new List<MiniYamlNode>();
+		List<MiniYamlNode> productionParadropNodes = new List<MiniYamlNode>();
+		List<MiniYamlNode> buildingBuildNodes = new List<MiniYamlNode>();
+		List<MiniYamlNode> buildingUndeployNodes = new List<MiniYamlNode>();
+
+		public override IEnumerable<string> BeforeUpdate(ModData modData)
+		{
+			// Reset state for each mod/map
+			paradropNodes.Clear();
+			ejectNodes.Clear();
+			productionParadropNodes.Clear();
+			buildingBuildNodes.Clear();
+			buildingUndeployNodes.Clear();
+
+			yield break;
+		}
+
+		string BuildMessage(string name, string replacement, List<MiniYamlNode> nodes)
+		{
+			var message = "The default value for " + name + " has been removed.\n"
+				+ "You may wish to explicitly define `" + replacement + "` at the following locations\n"
+				+ "if the sound has not already been inherited from a parent definition.\n";
+
+			foreach (var n in nodes)
```

Comment:
I should split out a helper for aggregating lines into sub-lists.  This is going to be done a lot.